backend/src/main/java/com/google/models/Timestamp.java

```diff
@@ -14,44 +14,42 @@
 
 package com.google.models;
 
-/** Wrapper for timestamp with day, month, year fields. */
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatterBuilder;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Wrapper for timestamp with day, month, year fields. 
+ * Accepts format in yyyy-MM-dd or yyyy-M-d or yyyy-MM-d or yyyy-M-dd, but prints format in yyyy-MM-dd.
+ */
 public final class Timestamp {
-  public static String TIMESTAMP_PROPERTY = "timestamp";
+  public static final String TIMESTAMP_PROPERTY = "timestamp";
   private static final String EXCEPTION_MESSAGE = "Invalid string format.";
-  private static int EXPECTED_STR_ARR_LENGTH = 3;
-
+  private static final DateTimeFormatter dateFormatter = new DateTimeFormatterBuilder()
+                                                    .appendOptional(DateTimeFormatter.ofPattern("yyyy-MM-dd"))
+                                                    .appendOptional(DateTimeFormatter.ofPattern("yyyy-M-d"))
+                                                    .appendOptional(DateTimeFormatter.ofPattern("yyyy-MM-d"))
+                                                    .appendOptional(DateTimeFormatter.ofPattern("yyyy-M-dd"))
+                                                    .toFormatter();
+
+  private final LocalDate date;
```

Comment:
(general comment for the file) what is this date for? is it different from the day + month + year?

Reply:
(change not necessary. i just don't know what it means haha)

Reply:
`LocalDate` is a pre-defined Java class that comes with many utility functions like checking whether `day`, `month`, `year`, is valid and converting `day`, `month`, `year` to string. 
I used this so i didnt have to manually check for whether the day, month, year passed in is valid.

Reply:
my bad I did not read the code close enough :') you're good!